entity-view/impl/src/main/java/com/blazebit/persistence/view/impl/metamodel/ManagedViewTypeImpl.java

```diff
@@ -345,6 +348,17 @@ public void run() {
                 }
             }
         }
+
+        Set<Class<? extends CTEProvider>> cteProviderClasses = viewMapping.getCteProviders();
+        if (cteProviderClasses != null) {
+            this.cteProviders = new LinkedHashSet<>();
```

Comment:
Always init the `cteProviders`, as we will have to add CTE providers from attribute subview types recursively.